### PR TITLE
Changes to support new boost-msvc14 package

### PR DESCRIPTION
**Pull request by yying**
This PR will contain changes needed to make provisioning script work better on Jenkins

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

File: CMakeLists.txt [hunk 1]
```diff
@@ -379,8 +379,8 @@ if(WINDOWS)
   # Win64 specific gflags
   set(BUILD_SHARED_LIBS ON CACHE BOOL "Build shared gtest library" FORCE)
 
-  include_directories("$ENV{HOMEDRIVE}/local/boost-msvc14/boost")
-  include_directories("$ENV{HOMEDRIVE}/local/boost-msvc14")
+  include_directories("${WINDOWS_DEP_DIR}/boost-msvc14/local/boost")
+  include_directories("${WINDOWS_DEP_DIR}/boost-msvc14/local")
   file(GLOB INCLUDE_DIRS LIST_DIRECTORIES true "${WINDOWS_DEP_DIR}/*/*/include")
```
Comment (theopolis) [inline CMakeLists.txt]: Doesn't this line handle boost now? So the above two lines can be removed?

**Comment by theopolis** [comment]
How about the linking logic for boost, does that need to be updated to handle the new within-choco install?

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

File: CMakeLists.txt (hunk 1 above)
Comment (yying) [inline CMakeLists.txt]: I don't think the boost install on Windows actually has an explicit `include/` directory

**Comment by theopolis** [comment]
ok to test

**Comment by osqueryer** [comment]
:+1: The commit 5b7e021ce261e5c7b7e50a6af989128819c25eb7 (Job results: [3019](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3019/)) was built successfully!

**Comment by osqueryer** [comment]
:+1: The commit 09468b1acc03bd727a7f8edfddc0a8c689ae838d (Job results: [3020](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3020/)) was built successfully!

**Pull request merged**